upgrade ubuntu and increase tested R version

barrettk opened the pull request:
upgrade ubuntu to 22.04 and increase a tested R version from R 4.1 to R 4.3 for the middle
 - Doesnt touch R versions for oldest or latest. 
 - We could bump oldest to R 4.1 to match MPN, though isnt necessary at this stage.

barrettk committed:
upgrade ubuntu and increase tested R version

 - doesnt touch R versions of oldest or latest

kyleam commented on .github/workflows/main.yaml:
```diff
@@ -20,12 +20,13 @@ jobs:
       matrix:
         config:
-          - os: ubuntu-20.04
+          - label: oldest
```
Why introduce label if it's not used anywhere downstream?

kyleam replied:
> I was just doing that for consistency across other packages. 

bbr has a job with an oldest label, but notice that it uses the label downstream.

I'm not aware of other packages that have a job marked with an oldest label.

> Thought we may have preferred that and just forgot to here. I can remove it if you prefer.

If it's not used, then I don't see a reason to add it.  (Keeping it would be like defining an unused variable in a script.)

kyleam commented on .github/workflows/main.yaml:
```diff
@@ -20,12 +20,13 @@ jobs:
-          - os: ubuntu-20.04
-            r: 4.1.3
+          - os: ubuntu-22.04
+            r: 4.3.1
```
When updating versions, my suggestion for the minimal set is the latest R version (`release`) and the two most recent R versions on the most recent Metworx (4.3.3 and 4.4.2).  That's a moving target, so the CI config for many of our packages will be stale and that's not a big deal (developers are running the full test suite on the latest Metworx before release, at the minimum).  But it provides a target once we get around to updating the CI for a package.

So the current minimum set here would be `release`, 4.4.2, and 4.3.3, with older versions on top as you like (so 4.0.5 based on the current config).

kyleam commented:
> We could bump oldest to R 4.1 to match MPN [...]

I'm not sure what this is referring to.  That latest snapshot targets the two most recent R versions on the latest Metworx (so R 4.3 and 4.4).

kyleam commented:
> @kyleam is there anything else I should add or adjust?

In this repo's settings, you should adjust the required checks to match the current set of CI jobs.

barrettk committed:
CI: update tested R versions

 - run with the two most recent R versions on the most recent Metworx (4.3.3 and 4.4.2)

kyleam reviewed (approved):
I think you should remove `label`, but that's not a blocker.

Ping me privately if you need me to point you to where to update the set of required checks.

barrettk committed:
ci: remove unused `oldest` label
